src/imars3d/backend/util/functions.py

```diff
@@ -14,3 +16,19 @@ def clamp_max_workers(max_workers: Union[int, None]) -> int:
     if max_workers is None:
         max_workers = 0
     return min(resource.RLIMIT_NPROC, max(1, multiprocessing.cpu_count() - 2)) if max_workers <= 0 else max_workers
+
+
+def to_time_str(value: datetime) -> str:
```

Comment:
Please set the default value to `datetime.now()` so all the other code can just call the `to_time_str()` without arguments. In my head it, it would simplify the other code.